lib/PhasarLLVM/Pointer/LLVMPointsToSet.cpp

```diff
@@ -59,6 +83,95 @@ LLVMPointsToSet::LLVMPointsToSet(ProjectIRDB &IRDB, bool UseLazyEvaluation,
   }
 }
 
+LLVMPointsToSet::LLVMPointsToSet(ProjectIRDB &IRDB,
+                                 const std::string& PointsToSetFile)
+{
+  load(PointsToSetFile, IRDB);
+}
+
+void LLVMPointsToSet::save(const std::string &PointsToSetFile, ProjectIRDB &IRDB) {
+  std::unordered_map<const llvm::Value *, unsigned long> ValueToIdMap;
+  std::ofstream OS(PointsToSetFile);
+
+
+  // ValueIds segment is just for debug. It will not been loaded.
+  OS << "[ValueIds]" << std::endl;
+  unsigned int NextID = 0;
+
+  // Traverse all values in the IRDB in a fixed order, and assign ID to each value.
+  traverseIRDB(IRDB, [&NextID, &ValueToIdMap, &OS](const llvm::Value *V) {
+    ValueToIdMap[V] = NextID;
+    OS << NextID << ": " << llvmIRToString(V) << std::endl;
+    ++NextID;
+  });
+
+  std::set<std::shared_ptr<std::unordered_set<const llvm::Value *>>> Printed;
+
+  OS << "[AnalyzedFunctions]" << std::endl;
+  for (auto &F : AnalyzedFunctions) {
+    OS << ValueToIdMap[F] << " ";
+  }
+  OS << std::endl;
+
+  OS << "[PointsToSets]" << std::endl;
+  for (auto &[_, S] : PointsToSets) {
+    if (Printed.find(S) != Printed.end()) {
+      // Print each set only once
+      continue;
+    }
+
+    Printed.insert(S);
+    for (const auto &V : *S) {
+      OS << ValueToIdMap[V] << " ";
+    }
+
+    OS << std::endl;
```

Comment:
```suggestion
    OS << '\n';
```
I guess, we don't need a flush here.